### PR TITLE
Rename Components and `export` them from `plugin-user-settings`

### PR DESCRIPTION
**Pull request by matteobarone**
Signed-off-by: Matteo Barone <matteo.barone@klarna.com>

## Hey, I just made a Pull Request!

Exported `General` component from the `@backstage/plugin-user-settings` plugin, to be able to use it in the consumer side and customize the `SettingPage`.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Hi!

It seems to me that the `General` component is both poorly named (if made part of a public interface), and of fairly little value since it's just a grid. I think it would potentially bring better value if you exported the things that are _in_ that grid, with nice naming that lets people compose their interface better. Does that make sense?

**Comment by matteobarone** [comment]
Hi @freben,
thanks for your feedback. I totally agree, I'll update this PR with something more smart.

**Comment by matteobarone** [comment]
@freben, I exported all the interesting components with the prefix `UserSettings*`, also I kept exporting the `General` (renamed in `UserSettingsGeneral`) component and exported the other two components that builds the SettingPage (maybe it is useful for someone that needs only one of them in the page, instead of all 3 of them).
What do you think, does it make sense?

File: .changeset/nice-bugs-beg.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-user-settings': patch
```
Comment (benjdlambert) [inline .changeset/nice-bugs-beg.md:2]: This is now a `minor` release as it has breaking changes.

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**